src/rules/mapreduce.jl

```diff
@@ -0,0 +1,30 @@
+#####
+##### `map`
+#####
+
+function rrule(::typeof(map), f, xs...)
+    T = Tuple{Core.Typeof(f), map(eltype, xs)...}
+    if hasmethod(rrule, T)
+        body = i->(ȳi, xis...)->begin
+            _, ∂x = rrule(f, xis...)
+            extern((∂x isa Tuple ? ∂x[i] : ∂x)(ȳi))
+        end
+    elseif hasmethod(frule, T)
```

Comment:
We could alternatively say that the `rrule` for `map(f, xs...)` requires that `f` have an `rrule`. That seems reasonable to me. I did it this way because that's what Nabla does: it does a `hasmethod` check to see whether there's a sensitivity defined, and if not, it uses a fallback FMAD implementation. Requiring an `rrule` for `f` would simplify this bizarre looking logic considerably.

Reply:
I've implemented that.